addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -158,12 +158,10 @@ def _generate_efaktur_invoice(self, delimiter):
 
             street = ', '.join([x for x in (move.partner_id.street, move.partner_id.street2) if x])
 
-            invoice_npwp = '000000000000000'
-            if commercial_partner.vat and len(commercial_partner.vat) >= 12:
-                invoice_npwp = commercial_partner.vat
-            elif (not commercial_partner.vat or len(commercial_partner.vat) < 12) and commercial_partner.l10n_id_nik:
-                invoice_npwp = commercial_partner.l10n_id_nik
-            invoice_npwp = invoice_npwp.replace('.', '').replace('-', '')
+            invoice_npwp = commercial_partner.vat.replace('.', '').replace('-', '')
+            etax_name = commercial_partner.l10n_id_tax_name or move.partner_id.name
```

Comment:
Still a change in behavior to confirm, sorry.

Before you would take the partner_id.name if '000000000000000', else you would take the etax name or partner_id.name
Here, it means that even if it is '000000000000000' you may take the etax name, which may not be what we want?

As we are changing it in stable, we need to be careful to not change the behavior more than necessary.

Reply:
confirmed with JGR on this one:
* always use l10n_id_tax_name first if exist, else the invoice customer's name =>etax_name
* if NPWP is zeros and they put NIK then format would become "{NIK}#NIK#NAMA#%{etax_name}"

with this, means my current implementation already suits the need